File: examples/async-await/async-await.swift
import Foundation

func sumParallel() async -> Int {
  async let a = f(100, 1)
  async let b = f(200, 5)
  return await a + b
}

func f(_ v: Int, _ s: Int) async -> Int
{
  print("Fetching \(v)...")
  // delay
  let ns = UInt64(s * 1_000_000_000)
  try! await Task.sleep(nanoseconds: ns)
  return v
}

@main struct Main {
  static func main() async {
    let result = await sumParallel()
    print(result)
  }
}

File: examples/async-await/async-await.rs
use tokio::time::{sleep, Duration};

async fn sum_parallel() -> u64 {
    let a_future = f(100, 1);
    let b_future = f(200, 5);
    let (a, b) = futures::join!(a_future, b_future);
    return a + b
}

async fn f(v: u64, s: u64) -> u64 {
    println!("Fetching {v}...");
    // delay
    sleep(Duration::from_secs(s)).await;
    return v
}

#[tokio::main]
async fn main() {
    let result = sum_parallel().await;
    print!("{}", result)
}